stdlib/public/core/PrintForDebugger.swift

```diff
@@ -0,0 +1,296 @@
+            willExpand = false
+          }
+        }
+      }
+    }
+
+    let count = Int(mirror.children.count)
+    let bullet = isRoot && (count == 0 || willExpand == false) ? ""
+      : count == 0    ? "- "
+      : maxDepth <= 0 ? "▹ " : "▿ "
+    print("\(bullet)", terminator: "", to: &targetStream)
+  
+    let collectionStatus = parentCollectionStatus.getChildStatus(child: mirror)
+  
+    if let nam = name {
+      print("\(nam) : ", terminator: "", to: &targetStream)
+    }
+
+    if let str = asStringRepresentation(value: value, mirror: mirror, count: count) {
+      print("\(str)", terminator: "", to: &targetStream)
+    }
+  
+    if (maxDepth <= 0) || !willExpand {
+      print("", to: &targetStream)
+      return
+    }
+
+    if let x = value {
+      if let valueIdentifier = asObjectIdentifier(value: x) {
+        if refsAlreadySeen.contains(valueIdentifier) {
+          print(" { ... }", to: &targetStream)
+          return
+        } else {
+          refsAlreadySeen.insert(valueIdentifier)
+        }
+      }
+    }
+
+    print("", to: &targetStream)
+  
+    var printedElements = 0
+  
+    if let sc = mirror.superclassMirror {
+      printForDebuggerImpl(value: nil,
+                           mirror: sc,
+                           name: "super",
+                           indent: indent + 2,
+                           maxDepth: maxDepth - 1,
+                           isRoot: false,
+                           parentCollectionStatus: .NotACollection,
+                           refsAlreadySeen: &refsAlreadySeen,
+                           maxItemCounter: &maxItemCounter,
+                           targetStream: &targetStream)
+    }
+  
+    for (optionalName,child) in mirror.children {
+      let childName = optionalName ?? "\(printedElements)"
+      if maxItemCounter <= 0 {
+        for _ in 0..<(indent+4) {
+          print(" ", terminator: "", to: &targetStream)
+        }
+        let remainder = count - printedElements
+        print("(\(remainder)", terminator: "", to: &targetStream)
+        if printedElements > 0 {
+          print(" more", terminator: "", to: &targetStream)
+        }
+        if remainder == 1 {
+          print(" child)", to: &targetStream)
+        } else {
+          print(" children)", to: &targetStream)
+        }
+        return
+      }
+    
+      printForDebuggerImpl(value: child,
+                           mirror: Mirror(reflecting: child),
+                           name: childName,
+                           indent: indent + 2,
+                           maxDepth: maxDepth - 1,
+                           isRoot: false,
+                           parentCollectionStatus: collectionStatus,
+                           refsAlreadySeen: &refsAlreadySeen,
+                           maxItemCounter: &maxItemCounter,
+                           targetStream: &targetStream)
+      printedElements += 1
+    }
+  }
+
+  struct StringOutput: OutputStream {
```

Comment:
`String` is an `OutputStream`, you don't need this wrapper struct.  Just print into a String instance.

```
(swift) var x: String = ""
// x : String = ""
(swift) print(42, to: &x)
(swift) x
// x : String = "42\n"
```